### ISSUE TITLE
Erorr in dbus

### ISSUE THREAD

**Issue by loustak** [Gerschtli/dwm-status#5]
Hello,

When I run `cargo run -- <config-file>` I get the following output before the program stop:
```
06:22:04 [ WARN] Config option 'debug' is deprecated and will be removed in 2.0.0. Log level is set to info by default.
06:22:04 [ INFO] Update all features
06:22:04 [ERROR] Error in process: reading process output failed (Os { code: 2, kind: NotFound, message: "No such file or directory" })
```

The first time I started it I had an additional error line saying `dbus error`

Thanks for help

**Comment by Gerschtli** [comment]
Hey, can you post your config file, preferably a minimal config file to reproduce the error?

Are all runtime dependencies needed for your enabled features available in your `PATH`? See according feature section in README.

**Comment by loustak** [comment]
I just run the default config file with the order field set:
```
{
  "debug": false,
  "order": ["backlight", "battery", "time"],
  "separator": " / ",
  "audio": {
    "control": "Master",
    "mute": "MUTE",
    "template": "S {VOL}%",
    "icons": []
  },
  "backlight": {
    "device": "intel_backlight",
    "template": "L {BL}%",
    "icons": []
  },
  "battery": {
    "charging": "▲",
    "discharging": "▼",
    "enable_notifier": true,
    "no_battery": "NO BATT",
    "notifier_critical": 10,
    "notifier_levels": [
      2,
      5,
      10,
      15,
      20
    ],
    "separator": " · ",
    "icons": []
  },
  "cpu_load": {
    "template": "{CL1} {CL5} {CL15}",
    "update_interval": 20
  },
  "network": {
    "no_value": "NA",
    "template": "{IPv4} · {IPv6} · {ESSID}"
  },
  "time": {
    "format": "%Y-%m-%d %H:%M",
    "update_seconds": false
  }
}
```

I found out that removing `"network"` from the order array solved the error. I'm gonna check that required lib are correctly installed.

**Comment by Gerschtli** [comment]
Could you please post the output of the following commands:

```
iwgetid -r
dig @resolver1.opendns.com A myip.opendns.com +short -4
dig @resolver1.opendns.com A myip.opendns.com +short -6
```

And does the following command remain open or exits it with an error?

```
ip monitor address link
```

**Comment by loustak** [comment]
I checked the necessary libs and I think they are installed on my arch linux. I'm not sure since arch doesn't probide directly `-lib` packages, but I don't get any errors related to that.

`iwgetid -r` correctly return the name of my internet box I'm connected at.
`dig @resolver1.opendns.com A myip.opendns.com +short -4` return an ipv4 adress.
`dig @resolver1.opendns.com A myip.opendns.com +short -6` return nothing.

`ip monitor address link` remains open.

Thanks for the help.

**Comment by Gerschtli** [comment]
Hey, could you try the branch [issue-5-fix](https://github.com/Gerschtli/dwm-status/tree/issue-5-fix) and post the error message?

And please post the output of:
```
dig @resolver1.opendns.com A myip.opendns.com +short
dig @resolver1.opendns.com AAAA myip.opendns.com +short
```

**Comment by loustak** [comment]
The branch solved the errors.
`dig @resolver1.opendns.com A myip.opendns.com +short` return 193.51.159.250
`dig @resolver1.opendns.com AAAA myip.opendns.com +short` return nothing

I was on the eduroam network know for IPV6 strange bugs...

**Comment by loustak** [comment]
I noticed a problem, if there was an error with the command used to gather the network ip the error is displayed in the status bar...

It printed in my status bar:
`server not found: connection timed out; no servers could be reached`

**Comment by Gerschtli** [comment]
Does the `dig` command exit with an exit code != 0? And is the message from stdout or stderr?

The time out is really hard to reproduce..

**Comment by loustak** [comment]
I found this way to make dig timeout:
`dig example.com @72.66.115.13` it's basically a blackhole server wich will just drop all requests. (source: https://serverfault.com/questions/776049/how-to-simulate-dns-server-response-timeout)

```
➜  ~ dig example.com @72.66.115.13

; <<>> DiG 9.14.5 <<>> example.com @72.66.115.13
;; global options: +cmd
;; connection timed out; no servers could be reached
➜  ~ echo $?
9
```

So, dig return an error code of 9 when there is no answer from the the server as described in the manual (https://linux.die.net/man/1/dig). The error message doesn't seems to be written to stderr.

**Comment by Gerschtli** [comment]
Ah okay, strange behaviour of dig. I will provide a fix based on the exit code later that day.

**Comment by Gerschtli** [comment]
Should be fixed by now in the issue branch. Can you verify it? If everything is working, I will merge and release it.

Thank you for reporting this issue and helping to improve this project! :)

**Comment by loustak** [comment]
Yep the issue is fixed! Thanks for helping me :)

**Comment by Gerschtli** [comment]
Fix is now available in version 1.6.3!
